server/sonar-web/src/main/js/apps/quality-gates/components/Projects.js

```diff
@@ -20,14 +20,105 @@
 import React from 'react';
 import escapeHtml from 'escape-html';
 import SelectList from '../../../components/SelectList';
+import SelectList2 from '../../../components/SelectList/SelectList';
 import { translate } from '../../../helpers/l10n';
 import { getBaseUrl } from '../../../helpers/urls';
+import {
+  searchGates,
+  associateGateWithProject,
+  dissociateGateWithProject
+} from '../../../api/quality-gates';
+
+/*::
+type State = {
+  projects: Array<{ id: number; name: string; selected: boolean }>
+};
+*/
 
 export default class Projects extends React.PureComponent {
+  state /*: State */ = {
+    projects: []
+  };
+
   componentDidMount() {
     this.renderSelectList();
+    this.handleSearch('', 'selected');
   }
 
+  handleSearch = (query /*: string*/, selected /*: string */) => {
+    const { qualityGate, organization } = this.props;
+    const requestData = {
+      gateId: qualityGate.id,
+      pageSize: 100,
+      selected
+    };
+
+    if (query !== '') {
+      requestData.query = query;
+    }
+
+    if (organization) {
+      requestData.organization = organization;
+    }
+
+    searchGates(requestData).then(
+      data => {
+        this.setState({ projects: data.results });
+      },
+      () => {}
+    );
+  };
+
+  handleSelect = (id /*: number*/) => {
+    const { qualityGate, organization } = this.props;
+    const requestData = {
+      gateId: qualityGate.id,
+      projectId: id
+    };
+
+    if (organization) {
+      requestData.organization = organization;
+    }
+
+    return associateGateWithProject(requestData).then(
+      () => {
+        this.setState(state => {
+          return {
+            projects: state.projects.map(project => {
+              return project.id === id ? { ...project, selected: true } : project;
+            })
+          };
+        });
+      },
+      () => {}
+    );
+  };
+
+  handleUnselect = (id /*: number*/) => {
+    const { qualityGate, organization } = this.props;
+    const requestData = {
+      gateId: qualityGate.id,
+      projectId: id
+    };
+
+    if (organization) {
+      requestData.organization = organization;
+    }
+
+    return dissociateGateWithProject(requestData).then(
+      () => {
+        this.setState(state => {
+          return {
+            projects: state.projects.map(project => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove curly braces and "return" from this arrow function body. ([javascript:S3524](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&pullRequest=3152&issues=AWIj7wLf7DpOGbPQ2n1C&open=AWIj7wLf7DpOGbPQ2n1C)

[ISSUE_KEY]: AWIj7wLf7DpOGbPQ2n1C